src/main/java/uk/gov/companieshouse/model/validator/CompanyNumberValidator.java

```diff
@@ -0,0 +1,65 @@
+package uk.gov.companieshouse.model.validator;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Pattern;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+import org.springframework.stereotype.Component;
+
+import uk.gov.companieshouse.config.CompanyNumberConfiguration;
+
+@Component
+public class CompanyNumberValidator implements ConstraintValidator<ValidCompanyNumber, String> {
+
+    private final Pattern prefixListRegex = Pattern.compile("(?i)^([A-Z][A-Z]?)\\b(,[A-Z][A-Z]?)*$");
+
+    private final Pattern companyNumberRegex;
+
+    public CompanyNumberValidator(CompanyNumberConfiguration prefixConfig) {
```

Comment:
what's the reason for having this constructor? Can we not add this logic to the `isValid` method?

Reply:
Constructor dependency injection. The program will fail earlier at runtime if the config is incorrect rather than when the method is called